src/monk/agent/database.c

```diff
@@ -34,6 +34,48 @@ PGresult* queryFileIdsForUploadAndLimits(fo_dbManager* dbManager, int uploadId,
       uploadId, left, right
     );
   }
+  else if(!ignoreIrre && scanFindings)
+  {
+    sql = g_strdup_printf("WITH allPfileData AS (SELECT DISTINCT pfile_fk FROM %s"
+      " WHERE upload_fk = $1 AND (ufile_mode&x'3C000000'::int) = 0 "
+      " AND (lft BETWEEN $2 AND $3) AND pfile_fk != 0) SELECT pfile_fk FROM allPfileData"
+      " WHERE pfile_fk NOT IN (SELECT pfile_fk FROM license_file WHERE rf_fk IN"
+      " (SELECT rf_pk FROM license_ref WHERE rf_shortname IN ('No_license_found', 'Void')));", tablename);
```

Comment:
SELECT can be optimized here
```suggestion
      " (SELECT rf_pk FROM license_ref WHERE rf_shortname = ANY(VALUES('No_license_found'), ('Void'))));", tablename);
```